src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/SetObjectTest.java

```diff
@@ -0,0 +1,134 @@
+                    try (PreparedStatement pstmt = con.prepareStatement("INSERT INTO "
+                            + AbstractSQLGenerator.escapeIdentifier(tableName) + " (id, dto) VALUES (?, ?)")) {
+                        pstmt.setInt(1, 1);
+                        pstmt.setObject(2, OffsetDateTime.parse(testValue));
+                        pstmt.executeUpdate();
+                    }
+
+                    try (ResultSet rs = stmt
+                            .executeQuery("SELECT COUNT(*) FROM " + AbstractSQLGenerator.escapeIdentifier(tableName)
+                                    + " WHERE id = 1 AND dto = '" + testValue + "'")) {
+                        rs.next();
+                        assertEquals(1, rs.getInt(1));
+                    }
+                } finally {
+                    TestUtils.dropTableIfExists(AbstractSQLGenerator.escapeIdentifier(tableName), stmt);
+                }
+            }
+        }
+    }
+
+    /**
+     * Tests setObject(n, java.time.OffsetTime.class).
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testSetObjectWithOffsetTime() throws SQLException {
+        try (Connection con = DriverManager.getConnection(connectionString)) {
+            final String testValue = "11:22:33.123456700+12:34";
+            final String expectedDto = "1970-01-01T" + testValue;
+            try (Statement stmt = con.createStatement()) {
+                stmt.executeUpdate("CREATE TABLE " + AbstractSQLGenerator.escapeIdentifier(tableName)
+                        + " (id INT PRIMARY KEY, dto DATETIMEOFFSET)");
+                try {
+                    try (PreparedStatement pstmt = con.prepareStatement("INSERT INTO "
+                            + AbstractSQLGenerator.escapeIdentifier(tableName) + " (id, dto) VALUES (?, ?)")) {
+                        pstmt.setInt(1, 1);
+                        pstmt.setObject(2, OffsetTime.parse(testValue));
+                        pstmt.executeUpdate();
+                    }
+
+                    try (ResultSet rs = stmt
+                            .executeQuery("SELECT COUNT(*) FROM " + AbstractSQLGenerator.escapeIdentifier(tableName)
+                                    + " WHERE id = 1 AND dto = '" + expectedDto + "'")) {
+                        rs.next();
+                        assertEquals(1, rs.getInt(1));
+                    }
+                } finally {
+                    TestUtils.dropTableIfExists(AbstractSQLGenerator.escapeIdentifier(tableName), stmt);
+                }
+            }
+        }
+    }
+
+    /**
+     * Tests setObject(n, java.time.OffsetTime.class) when 'setSendTimeAsDatetime' connection property is false.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testSetObjectWithOffsetTime_sendTimeAsDatetimeDisabled() throws SQLException {
+        try (Connection con = DriverManager.getConnection(connectionString)) {
+            ((SQLServerConnection) con).setSendTimeAsDatetime(false);
```

Comment:
Changing 'setSendTimeAsDatetime' connection property has no impact on OffsetDateTime and OffsetTime java types or underlying column DATETIMEOFFSET. Is there a specific reason to test with this property change?

Reply:
Thanks for the review, @cheenamalhotra !
Note the difference between `expectedDto`s.
The connection property affects the 'date' part supplemented by the driver when inserting `java.time.OffsetTime` into `DATETIMEOFFSET` column.
I should mention that this behavior is ported from the original code. Please see the following line.
https://github.com/Microsoft/mssql-jdbc/pull/831/files#diff-452e970f57e6bb485626382755aeb835L635